RobotCode/snobot2018/src/org/snobot/autonomous/CommandParser.java

```diff
@@ -0,0 +1,147 @@
+    }
+
+    /**
+     * Takes a list of Strings and creates a Command.
+     * 
+     * @param args
+     *            The command's name and parameters.
+     */
+    @Override
+    protected Command parseCommand(List<String> aCommandArgs)
+    {
+        String commandName = aCommandArgs.get(0);
+        Command newCommand = null;
+        try
+        {
+            // Generic from base class
+            switch (commandName)
+            {
+            case AutonomousCommandNames.sPARALLEL_COMMAND:
+            {
+                newCommand = parseParallelCommand(aCommandArgs);
+                break;
+            }
+            case AutonomousCommandNames.sWAIT_COMMAND:
+            {
+                newCommand = parseWaitCommand(aCommandArgs);
+                break;
+            }
+            default:
+            {
+                // Subsystems from AuotonomousoCommandCreator
+                newCommand = mCommandCreator.createCommand(commandName, aCommandArgs, mSnobot);
+                if (null == newCommand)
+                {
+                    addError("Received unexpected command name '" + commandName + "'");
+                }
+                break;
+            }
+            }
+        }
+        catch (IndexOutOfBoundsException ex)
+        {
+            addError("You have not specified enough aguments for the command: " + commandName + ". " + ex.getMessage());
+        }
+        catch (Exception ex)
+        {
+            addError("Failed to parse the command: " + commandName + ". " + ex.getMessage());
+            sLOGGER.log(Level.ERROR, "Failed to parse the command: " + commandName + ". ", ex);
+        }
+        return newCommand;
+    }
+
+    protected Command parseWaitCommand(List<String> aArgs)
+    {
+        double time = Double.parseDouble(aArgs.get(1));
+        return new WaitCommand(time);
+    }
+
+    @Override
+    public CommandGroup readFile(String aFilePath) // NOPMD
+    {
+        if (aFilePath == null)
+        {
+            aFilePath = "NOT FOUND!";
+        }
+
+        mAutonFilenameEntry.setString(aFilePath);
+        return super.readFile(aFilePath);
+    }
+
+    /**
+     * Saves the autonomous mode to the RoboRio.
+     */
+    public void saveAutonMode()
+    {
+        String newText = mAutonSdTableTextName.getString("");
+        String filename = mAutonFilenameEntry.getString("auton_file.txt");
+
+        sLOGGER.log(Level.INFO, "\n"
+                + "*****************************************\n"
+                + "Saving auton mode to " + filename + "\n"
+                + "*****************************************");
+
+        try
+        {
+            BufferedWriter bw = new BufferedWriter(new FileWriter(new File(filename)));
+            bw.write(newText);
+            bw.close();
+        }
+        catch (Exception ex)
+        {
+            sLOGGER.log(Level.INFO, ex);
+        }
+        // sAUTON_TABLE.putBoolean(SmartDashboardNames.sSAVE_AUTON, false);
```

Comment:
Either put this in (i think it is important for the SD file updating) or remove the line